### PR TITLE
Update shader docs

### PR DESCRIPTION
**Pull request by Chaosus**
I carefully checked original shader_types.cpp and found some errors in documentation of shader language.. which I tried to fix in this PR.
First, I've added new built-in functions(which added recently by me) and reformat other functions. 
Second, I refactor all builtins and added "in"(for input constant) and "out"(for others) in places where it misses, and replace "inout" to simply "out" because it sounds confusing cuz in shader_types.cpp there is no inout mark for these variables.(I could back, just want to review from others).

### CODE REVIEW

File: tutorials/shading/shading_language.rst [hunk 1]
```diff
@@ -947,13 +954,13 @@ Vertex Built-Ins
...
+| out vec2 **VERTEX**            | Vertex in image space.                                         |
 +--------------------------------+----------------------------------------------------------------+
-| inout vec2 **UV**              | UV.                                                            |
+| out vec2 **UV**                | UV.                                                            |
 +--------------------------------+----------------------------------------------------------------+
-| inout vec4 **COLOR**           | Color from vertex primitive.                                   |
+| out vec4 **COLOR**             | Color from vertex primitive.                                   |
 +--------------------------------+----------------------------------------------------------------+
-| out vec2 **POINT_SIZE**        | Point size for point drawing.                                  |
+| out float **POINT_SIZE**       | Point size for point drawing.                                  |
```
Comment (StraToN) [inline tutorials/shading/shading_language.rst]: So POINT_SIZE was converted from `vec2` to `float`? Just asking for confirmation, I have no idea.

Comment (Chaosus) [reply tutorials/shading/shading_language.rst]: Yeah

Comment (Chaosus) [reply tutorials/shading/shading_language.rst]: According to shader_types.cpp it have TYPE_FLOAT, some types was also incorrect and I changed them

**Comment by Chaosus** [comment]
+I also added the simple matrix construction/accessing description

**Pull request merged**